solr/solrj/src/java/org/apache/solr/common/cloud/Replica.java

```diff
@@ -103,31 +104,41 @@ public enum Type {
      * support NRT (soft commits) and RTG. Any {@link Type#NRT} replica can become a leader. A shard
      * leader will forward updates to all active {@link Type#NRT} and {@link Type#TLOG} replicas.
      */
-    NRT(true),
+    NRT(true, CollectionAdminParams.NRT_REPLICAS),
     /**
      * Writes to transaction log, but not to index, uses replication. Any {@link Type#TLOG} replica
      * can become leader (by first applying all local transaction log elements). If a replica is of
      * type {@link Type#TLOG} but is also the leader, it will behave as a {@link Type#NRT}. A shard
      * leader will forward updates to all active {@link Type#NRT} and {@link Type#TLOG} replicas.
      */
-    TLOG(true),
+    TLOG(true, CollectionAdminParams.TLOG_REPLICAS),
     /**
      * Doesn’t index or writes to transaction log. Just replicates from {@link Type#NRT} or {@link
      * Type#TLOG} replicas. {@link Type#PULL} replicas can’t become shard leaders (i.e., if there
      * are only pull replicas in the collection at some point, updates will fail same as if there is
      * no leaders, queries continue to work), so they don’t even participate in elections.
      */
-    PULL(false);
+    PULL(false, CollectionAdminParams.PULL_REPLICAS);
 
     public final boolean leaderEligible;
```

Comment:
But on the other it would require yet another look in `Replica.Type#defaultType()` or static caching. Have you seen other places where it would be useful? If it's only in `CollectionHandlingUtils#makeCollectionPropsAndDefaults`, I would prefer not to optimise now for a single use case.

Reply:
Not optimization. Intent is to improve code readability

Reply:
Not sure of the readability improvement between `x.default` or `x == Type.defaultValue()`. The former is more concise, but it is a very small improvement imo.